pkgs/applications/misc/mangal/default.nix

```diff
@@ -0,0 +1,35 @@
+{ lib, stdenv, buildGoModule, fetchFromGitHub, installShellFiles, go_1_18, lua }:
+
+buildGoModule rec {
+  pname = "mangal";
+  version = "3.0.3";
+
+  src = fetchFromGitHub {
+    owner = "metafates";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-+U3KDIfpHWzWJrw0DoF1wySwLfFNvknPBUEhl6IowFo=";
+  };
+
+  vendorSha256 = "sha256-PrslSjWkiLJiHnJFuIz2+RU7S446ahcf31nQ6GYfAhU=";
+
+  ldflags = [ "-s" "-w" ];
+
+  nativeBuildInputs = [ installShellFiles ];
+
+  postInstall = ''
```

Comment:
```suggestion
  postInstall = lib.optionalString (stdenv.hostPlatform == stdenv.buildPlatform) ''
```

Reply:
This is required, as explained in the linked comment.

Reply:
Well, not really hard-required, but why break cross-compilation when it can be supported from the start.

And, yes, the cross-compiled package will have a different output, but it cannot really be avoided, if to generate the completion you need to run the compiled `$out/bin/mangal` binary. I am kinda confused why you say it works.

Reply:
Basically I don't know if the cross-compiling build environment also includes by default qemu to run the target application.
If it doesn't then your change is necessary to not break cross compilation. If it is, then it would just work.
In my case qemu is enabled in NixOS using `boot.binfmt.emulatedSystems = [ "armv7l-linux" "aarch64-linux" ];` in my config and the binary can be executed during the build and even after.

Reply:
I trust the commenter in the link above requesting the fix for cross-compilation in another PR, who is a code owner of the Go compiler.

Reply:
Ok, I updated the derivation as suggested. I'm still not that expert in Nix, that's why I'm often asking for more details